test/utils/ktesting/example_test.go

```diff
@@ -0,0 +1,59 @@
+//go:build example
+// +build example
+
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ktesting_test
+
+// The tests below will fail and therefore are excluded from
+// normal "make test" via the "example" build tag. To run
+// the tests and check the output, use "go test -tags example ."
+
+import (
+	"testing"
+
+	"k8s.io/kubernetes/test/utils/ktesting"
+)
+
+func TestError(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	tCtx.Error("some", "thing")
+}
+
+func TestErrorf(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	tCtx.Errorf("some %s", "thing")
+}
+
+func TestFatal(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	tCtx.Fatal("some", "thing")
+	tCtx.Log("not reached")
+}
+
+func TestFatalf(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	tCtx.Fatalf("some %s", "thing")
+	tCtx.Log("not reached")
+}
+
+func TestInfo(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	tCtx.Log("hello via Log")
+	tCtx.Logger().Info("hello via Info")
+	tCtx.Error("some", "thing")
+}
```

Comment:
Here's the current output:

```console
go test -tags example .
--- FAIL: TestError (0.00s)
    example_test.go:34: ERROR: some thing
--- FAIL: TestErrorf (0.00s)
    example_test.go:39: ERROR: some thing
--- FAIL: TestFatal (0.00s)
    example_test.go:44: ERROR: some thing
--- FAIL: TestFatalf (0.00s)
    example_test.go:50: ERROR: some thing
--- FAIL: TestInfo (0.00s)
    example_test.go:56: hello via Log
    example_test.go:57: I1226 14:05:00.662639] hello via Info
    example_test.go:58: ERROR: some thing
FAIL
FAIL	k8s.io/kubernetes/test/utils/ktesting	0.023s
```

Should all output go through the logger, i.e. have the date and time prefix?

I've always been wondering whether tests should use klog or `testing.T.Log`. It's not done consistently today. If both use the logger, the question becomes mute - use whatever is more convenient, the result will be the same.